### ISSUE TITLE
repeated test runs in same session may raise warnings

### ISSUE THREAD

**Issue by maxheld83** [r-lib/roxygen2#1189]
As of https://github.com/r-lib/roxygen2/commit/b3053b694399416543af877837efb29e6ec34d1f, repeated tests with some failing tests in the same session may raise these warnings:

```
1: In .Internal(gc(verbose, reset, full)) :
  closing unused connection 7 (/var/folders/1f/9tm8hrm179b6qsv8mw9lbv580000gn/T//RtmpWHa5Ey/file92741d50bb5e)
2: In .Internal(gc(verbose, reset, full)) :
  closing unused connection 6 (/var/folders/1f/9tm8hrm179b6qsv8mw9lbv580000gn/T//RtmpWHa5Ey/file9274512de9cd)
3: In .Internal(gc(verbose, reset, full)) :
  closing unused connection 5 (/var/folders/1f/9tm8hrm179b6qsv8mw9lbv580000gn/T//RtmpWHa5Ey/file92746b389a99)
4: In .Internal(gc(verbose, reset, full)) :
  closing unused connection 4 (/var/folders/1f/9tm8hrm179b6qsv8mw9lbv580000gn/T//RtmpWHa5Ey/file92743e1b3129)
```

The exact number of closed unused connections varies somewhat.

To reproduce:

1. Introduce a failing test somewhere:
    ```r
     test_that("trigger unused connections warning",  testthat::fail())
    ```
2. Run the test at least twice:
    ```r
    devtools::test()
    devtools::test()
    ```

This isn't really a big problem, though it can get contributors off on the wrong path if their tests break during development.
(I thought I might must have somehow mishandled connections in https://github.com/r-lib/roxygen2/pull/1182).

Based on the paths, my guess would be that these warnings are caused by connections to `tempfile()`  in tests, which somehow may not be properly closed in time.

Maybe something like this:

https://github.com/r-lib/testthat/blob/f1dd05cbae2432c1ad8a3db53849ce01819f4172/tests/testthat/test-expect-known.R#L13-L20

`session_info()`:

```
─ Session info ───────────────────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 4.0.3 (2020-10-10)
 os       macOS Big Sur 10.16         
 system   x86_64, darwin17.0          
 ui       RStudio                     
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       Europe/Berlin               
 date     2021-01-19                  

─ Packages ───────────────────────────────────────────────────────────────────────────────
 ! package          * version     date       lib source                            
   assertthat         0.2.1       2019-03-21 [1] CRAN (R 4.0.2)                    
   brew               1.0-6       2011-04-13 [1] CRAN (R 4.0.2)                    
   callr              3.5.1       2020-10-13 [1] CRAN (R 4.0.2)                    
   cli                2.2.0       2020-11-20 [1] CRAN (R 4.0.2)                    
   codetools          0.2-18      2020-11-04 [1] CRAN (R 4.0.2)                    
   commonmark         1.7         2018-12-01 [1] CRAN (R 4.0.2)                    
   crayon             1.3.4       2017-09-16 [1] CRAN (R 4.0.2)                    
   desc               1.2.0       2018-05-01 [1] CRAN (R 4.0.2)                    
   devtools         * 2.3.2       2020-09-18 [1] CRAN (R 4.0.2)                    
   diffobj            0.3.3       2021-01-07 [1] CRAN (R 4.0.2)                    
   digest             0.6.27      2020-10-24 [1] CRAN (R 4.0.2)                    
   ellipsis           0.3.1       2020-05-15 [1] CRAN (R 4.0.2)                    
 R empty            * 1.0.0       <NA>       [?] <NA>                              
   evaluate           0.14        2019-05-28 [1] CRAN (R 4.0.1)                    
   fansi              0.4.2       2021-01-15 [1] CRAN (R 4.0.2)                    
   fs                 1.5.0       2020-07-31 [1] CRAN (R 4.0.2)                    
   glue               1.4.2       2020-08-27 [1] CRAN (R 4.0.2)                    
   htmltools          0.5.1.9000  2021-01-13 [1] Github (rstudio/htmltools@11cfbf3)
   knitr              1.30        2020-09-22 [1] CRAN (R 4.0.2)                    
   lifecycle          0.2.0       2020-03-06 [1] CRAN (R 4.0.2)                    
   magrittr           2.0.1       2020-11-17 [1] CRAN (R 4.0.2)                    
   memoise            1.1.0       2017-04-21 [1] CRAN (R 4.0.2)                    
   muggle           * 0.1.0-3     2021-01-13 [1] Github (subugoe/muggle@94373e6)   
   pillar             1.4.7       2020-11-20 [1] CRAN (R 4.0.2)                    
   pkgbuild           1.2.0       2020-12-15 [1] CRAN (R 4.0.2)                    
   pkgconfig          2.0.3       2019-09-22 [1] CRAN (R 4.0.2)                    
   pkgload            1.1.0       2020-05-29 [1] CRAN (R 4.0.2)                    
   prettyunits        1.1.1       2020-01-24 [1] CRAN (R 4.0.2)                    
   processx           3.4.5       2020-11-30 [1] CRAN (R 4.0.2)                    
   ps                 1.5.0       2020-12-05 [1] CRAN (R 4.0.2)                    
   purrr              0.3.4       2020-04-17 [1] CRAN (R 4.0.2)                    
   R.methodsS3        1.8.1       2020-08-26 [1] CRAN (R 4.0.2)                    
   R.oo               1.24.0      2020-08-26 [1] CRAN (R 4.0.2)                    
   R6                 2.5.0       2020-10-28 [1] CRAN (R 4.0.2)                    
   Rcpp               1.0.6       2021-01-15 [1] CRAN (R 4.0.2)                    
   rematch2           2.1.2       2020-05-01 [1] CRAN (R 4.0.2)                    
   remotes            2.2.0       2020-07-21 [1] CRAN (R 4.0.2)                    
   rlang              0.4.10.9000 2021-01-18 [1] Github (r-lib/rlang@321a057)      
   rmarkdown          2.6         2020-12-14 [1] CRAN (R 4.0.2)                    
 P roxygen2         * 7.1.1.9000  2021-01-15 [?] local                             
   rprojroot          2.0.2       2020-11-15 [1] CRAN (R 4.0.2)                    
   rstudioapi         0.13        2020-11-12 [1] CRAN (R 4.0.2)                    
   sessioninfo        1.1.1       2018-11-05 [1] CRAN (R 4.0.2)                    
   stringi            1.5.3       2020-09-09 [1] CRAN (R 4.0.2)                    
   stringr            1.4.0       2019-02-10 [1] CRAN (R 4.0.2)                    
 R testEagerData    * 0.1         <NA>       [?] <NA>                              
 R testNamespace    * 0.1         <NA>       [?] <NA>                              
 R testNonASCII     * 0.1         <NA>       [?] <NA>                              
 R testRbuildignore * 0.1         <NA>       [?] <NA>                              
   testthat         * 3.0.1       2020-12-17 [1] CRAN (R 4.0.2)                    
 R testUtf8Escape   * 0.1         <NA>       [?] <NA>                              
   tibble             3.0.5       2021-01-15 [1] CRAN (R 4.0.2)                    
   usethis          * 2.0.0       2020-12-10 [1] CRAN (R 4.0.2)                    
   vctrs              0.3.6       2020-12-17 [1] CRAN (R 4.0.2)                    
   waldo              0.2.3       2020-11-09 [1] CRAN (R 4.0.2)                    
   withr              2.4.0       2021-01-16 [1] CRAN (R 4.0.2)                    
   xfun               0.20        2021-01-06 [1] CRAN (R 4.0.2)                    
   xml2               1.3.2       2020-04-23 [1] CRAN (R 4.0.2)                    
   yaml               2.2.1       2020-02-01 [1] CRAN (R 4.0.2)                    

[1] /Library/Frameworks/R.framework/Versions/4.0/Resources/library

 P ── Loaded and on-disk path mismatch.
 R ── Package was removed from disk.
```

**Comment by maxheld83** [comment]
it appears that this cannot be reproduced inside the build pane in RSTudio, but only when running `testthat::test()` in the console.

Maybe the build pane will always start a new process and/or suppresses warnings.

**Comment by maxheld83** [comment]
~~this is closed in #1191~~ (?!?)

Well, at least I can no longer reproduce the warnings in #1191 😼.
